kp_test.go

```diff
@@ -2670,6 +2670,115 @@ func TestCancel_DualAuthDelete(t *testing.T) {
 	assert.True(t, gock.IsDone(), "Expected HTTP requests not called!")
 }
 
+func TestCreateKeyWithAliases(t *testing.T){
+	defer gock.Off()
+	aliases := []string{"alias1", "alias2", "alias3"}
+	keyName := "test secret with alias"
+	keyResponse := []byte(`{
+		"metadata": {
+			"collectionTotal": 1,
+			"collectionType": "application/vnd.ibm.kms.key+json"
+		},
+		"resources": [
+			{
+				"algorithmType": "AES",
+				"aliases": [
+					"alias1",
+					"alias2",
+					"alias3"
+				],
+				"createdBy": "xyz",
+				"creationDate": "2020-11-06T22:40:44Z",
+				"crn": "dummycrn:v1:staging:public",
+				"deleted": false,
+				"extractable": false,
+				"id": "7b07-40b4-a306-eb98ee",
+				"lastUpdateDate": "2020-11-06T22:40:44Z",
+				"name": "test secret with alias",
+				"state": 1,
+				"type": "application/vnd.ibm.kms.key+json"
+			}
+		]
+	}`)
+
+	gock.New("http://example.com").
+		Post("/api/v2/keys").
+		Reply(201).
+		Body(bytes.NewReader(keyResponse))
+
+	c, _, err := NewTestClient(t, nil)
+	gock.InterceptClient(&c.HttpClient)
+	defer gock.RestoreClient(&c.HttpClient)
+	c.tokenSource = &FakeTokenSource{}
+
+	key, err := c.CreateKeyWithAliases(context.Background(), keyName, nil, false, aliases)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, key)
+	assert.Equal(t, key.Name, keyName)
+	assert.Equal(t, key.Aliases, aliases)
+	assert.False(t, key.Extractable)
+
+	assert.True(t, gock.IsDone(), "Expected HTTP requests not called!")
+}
+
+func TestCreateImportedKeyWithAliases(t *testing.T) {
+	defer gock.Off()
+	aliases := []string{"importedAlias1","importedAlias2","importedAlias3","importedAlias4","importedAlias5"}
+	keyName := "importeKeyWithAliases"
+	payload := "108v8jH6ZGGs/ekY/JRz4iy8hiDicoTi1n4vnfK9tsI="
+	keyResponse := []byte(`{
+		"metadata": {
+			"collectionTotal": 1,
+			"collectionType": "application/vnd.ibm.kms.key+json"
+		},
+		"resources": [
+			{
+				"id": "4b3d-6bce-9a29-1ca9a",
+				"name": "importeKeyWithAliases",
+				"type": "application/vnd.ibm.kms.key+json",
+				"aliases": [
+						"importedAlias1",
+						"importedAlias2",
+						"importedAlias3",
+						"importedAlias4",
+						"importedAlias5"
+				],
+				"algorithmType": "AES",
+				"createdBy": "abc-xyz",
+				"creationDate": "2015-05-12T06:32:30Z",
+				"lastUpdateDate": "2015-05-12T06:32:30Z",
+				"extractable": true,
+				"payload": "108v8jH6ZGGs/ekY/JRz4iy8hiDicoTi1n4vnfK9tsI=",
+				"state": 1,
+				"crn": "dummycrn:v1:bluemix:public:kms:",
+				"deleted": false
+			}
+		]
+	}`)
+	
+	gock.New("http://example.com").
+		Post("/api/v2/keys").
+		Reply(201).
+		Body(bytes.NewReader(keyResponse))
+
+	c, _,err := NewTestClient(t, nil)
+	gock.InterceptClient(&c.HttpClient)
+	defer gock.RestoreClient(&c.HttpClient)
+	c.tokenSource = &FakeTokenSource{}
+	
+	key, err := c.CreateImportedKeyWithAliases(context.Background(), keyName, nil, payload, "", "", true, aliases)
```

Comment:
Main point for adding this test was to test with encryptedNonce and iv. But you are skipping it.

Reply:
done